### PR TITLE
Performance optimization

### PR DESCRIPTION
**Pull request by sudarshann**
Includes the code optimization for the cron job performance issue.

Description
--
The Code that fetched the vendor using model during the loop at different stages have been updated to use the object that was prefetched. Also minor bugs in email template was also fixed

### Fixed Issues (if relevant)
1. Mowdirect/stock_module#10: Performance Issue with cron Job
2. Mowdirect/stock_module#7: Comprehensibility Issue 

### Manual testing scenarios
1. Compare and monitor the CPU usaged compared to the previous version of the code using blackfire.io or server cpu monitor

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/local/Mowdirect/Emailimporter/Helper/GmailConnect.php [hunk 1]
```diff
@@ -1,9 +1,13 @@
 <?php
 
+ini_set('display_errors', 1);
+error_reporting(E_ALL);
+
 class Mowdirect_Emailimporter_Helper_GmailConnect extends Mage_Core_Helper_Abstract {
 
     private $token;
     private $client;
+    private $redirectUri = 'http://127.0.0.1/index.php/email_importer/index/gmailredirect/';
```
Comment (waynetheisinger) [inline app/code/local/Mowdirect/Emailimporter/Helper/GmailConnect.php]: @sudarshann The http://127.0.0.1 URL isn't going to work with kubernetes use something similar to this https://stackoverflow.com/questions/11450146/magento-generating-url-for-a-backend-action-with-key

File: app/code/local/Mowdirect/Emailimporter/Helper/GmailConnect.php [hunk 2]
```diff
@@ -1,9 +1,13 @@
 <?php
 
+ini_set('display_errors', 1);
```
Comment (waynetheisinger) [inline app/code/local/Mowdirect/Emailimporter/Helper/GmailConnect.php]: @sudarshann remove for production

File: app/code/local/Mowdirect/Emailimporter/Helper/GmailConnect.php [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+error_reporting(E_ALL);
```
Comment (waynetheisinger) [inline app/code/local/Mowdirect/Emailimporter/Helper/GmailConnect.php]: @sudarshann remove for production

File: app/code/local/Mowdirect/Emailimporter/Helper/Vendor.php [hunk 4]
```diff
@@ -1,7 +1,15 @@
 <?php
 
 class Mowdirect_Emailimporter_Helper_Vendor extends Mage_Core_Helper_Abstract {
-    
+
+    public function get_vendors() {
+        $vendors = Mage::getModel('udropship/vendor')->getCollection()->getData();
+        foreach ($vendors as $key => $vendor){
+           $custom_data = json_decode($vendor['custom_vars_combined'], true);
```
Comment (waynetheisinger) [inline app/code/local/Mowdirect/Emailimporter/Helper/Vendor.php:8]: @sudarshann was `custom_vars_combined` already part of the model?

File: app/code/local/Mowdirect/Emailimporter/Model/Observer.php [hunk 5]
```diff
@@ -5,48 +5,56 @@ class Mowdirect_Emailimporter_Model_Observer {
 
     public function handleEmailCron() {
         Mage::log('Stock importer (info):  cron started ');
-        $vendors = Mage::helper('emailimporter')->get_vendors();
+        $vendors = Mage::helper('emailimporter/Vendor')->get_vendors();
+        $vendor_model = Mage::getModel('udropship/vendor');
 
-        foreach ($vendors as $vendor) {
+        $vendor_to_update = new Varien_Data_Collection();
```
Comment (waynetheisinger) [inline app/code/local/Mowdirect/Emailimporter/Model/Observer.php]: @sudarshann, this is a generic collection so for code readability it should be `$vendors_to_update` i.e. the plural of vendors

**Review by waynetheisinger — COMMENTED** [review]
@sudarshann left a few minor comments to be looked at before merge

File: app/code/local/Mowdirect/Emailimporter/Helper/Vendor.php (hunk 4 above)
Comment (sudarshann) [reply app/code/local/Mowdirect/Emailimporter/Helper/Vendor.php:8]: Yes Wayne this already part of the model. It was kept as json encoded. thats why we need a decoding of it before processing the vendor loop.

**Pull request merged**